### ISSUE TITLE
TextEdit ranges use UTF-16 code units — incorrect positions for non-ASCII content

### ISSUE THREAD

**Issue by saphryn** [markuplint/markuplint#3892]
## Bug Description

`TextEdit.range` is defined as `[start, end)` offsets in **UTF-16 code units**. JavaScript strings use UTF-16 internally, which makes this work correctly within the markuplint TS codebase. However, any external tool consuming markuplint's fix output — IDE integrations, code formatters, other linters — almost universally uses **UTF-8 byte offsets** or **Unicode code point indices**. This causes incorrect fix positions for any content containing non-ASCII characters.

## Root Cause

In `packages/@markuplint/ml-config/src/types.ts`:

```typescript
export type TextEdit = {
  /** 0-based character offsets (UTF-16 code units): [start, end) */
  readonly range: readonly [start: number, end: number];
  readonly text: string;
};
```

The comment explicitly documents UTF-16, but:
1. Most text editors and language servers use UTF-8 byte offsets (LSP specification)
2. ESLint's `TextEdit` uses UTF-16 code units, which is the same choice markuplint made — but ESLint's ecosystem understands this convention
3. markuplint has no equivalent ecosystem warning or conversion utility

## Impact

- **Incorrect fixes for Cyrillic, CJK, Arabic, emoji**: An element at byte offset 100 in UTF-8 might be at code unit 50 in UTF-16 (if each Cyrillic letter is 2 bytes → 1 code unit, but emoji are 4 bytes → 2 code units). Applying a UTF-16-based fix to a UTF-8 stream produces corrupted markup.
- **Broken IDE integration**: Any VS Code extension or LSP-based tool that consumes markuplint's `TextEdit` output will misalign fixes for non-ASCII content.
- **Silent corruption**: The fix appears to apply but the replacement lands on the wrong character range.

## Reproduction

1. Lint a file containing Cyrillic text: `<p>Привет мир</p>`
2. Trigger an autofix that modifies a token near the Cyrillic text
3. Observe: the fix lands on wrong character offsets in UTF-8 tools

## Proposed Fix

**Option A**: Change `TextEdit.range` to use UTF-8 byte offsets and update all rule fix callbacks accordingly. This matches LSP and most external tool expectations.

**Option B**: Keep UTF-16 internally but provide a conversion utility (`utf16ToUtf8ByteOffset(text, utf16Offset)`) and document the encoding clearly in the API.

**Option C**: At minimum, add a prominent warning in the documentation and in the `FixSummary` type that `TextEdit.range` uses UTF-16 code units, so consumers know to convert.

## Additional Context

The Rust v6 rewrite (https://github.com/markuplint/markuplint/issues/3178) has an opportunity to define the fix output API with UTF-8 as the default, avoiding this compatibility trap.

Affected files: `packages/@markuplint/ml-config/src/types.ts`, any rule implementing fix callbacks

**Comment by YusukeHirao** [comment]
Thanks for the report, but after verification the premise turned out to be incorrect, so I'm closing this one.

**The LSP specification's position encoding is UTF-16 by default, not UTF-8.** Prior to LSP 3.17, UTF-16 was the only supported encoding; LSP 3.17 added the `positionEncoding` capability whose default is still `'utf-16'`. UTF-8 only applies when a client and server explicitly negotiate it.

markuplint's pipeline is consistently UTF-16 end to end:

- Offsets are produced with `String#length` (`@markuplint/parser-utils` `get-location.ts`), which counts UTF-16 code units.
- Fixes are applied with `String#slice` (`@markuplint/ml-core` `fix-applier.ts`), which indexes by UTF-16 code units — so `TextEdit.range` applies correctly by construction.
- The VS Code extension converts offsets to LSP positions via `offsetToPosition` (`vscode/src/server/code-actions.ts`), producing UTF-16-based `character` values, which is exactly what LSP expects.
- A multibyte regression test already exists in `fix-applier.spec.ts` (`こんにちは世界`).

ESLint's fix ranges use the same UTF-16 code-unit convention. External consumers that operate in UTF-8 byte offsets need to convert regardless of which JS-ecosystem linter they consume — that is a property of the ecosystem convention, not a bug in markuplint.

The suggestion to document the encoding is already satisfied: the `TextEdit` type's JSDoc explicitly states "0-based character offsets (UTF-16 code units): [start, end)".
